src/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -215,9 +217,128 @@ public SAMSequenceRecord addSequenceAlias(final String originalName,
+        if(region.isEmpty())  throw new IllegalArgumentException("parseInterval: region is empty");
+        final int colon = region.indexOf(':');
+        // no colon, chromosome only
+        if(colon == -1) {
+            final SAMSequenceRecord record = getSequence(region);
+            if(record == null) throw new IllegalArgumentException("Unknown chromosome \""+region+"\".");
+            // return whole chromosome
+            return new Interval(
+                    record.getSequenceName(),
+                    1,
+                    record.getSequenceLength()
+                    );
+        }
+        if(colon == 0)  throw new IllegalArgumentException("Empty chromosome in "+region);
+        final SAMSequenceRecord record = getSequence(region.substring(0,colon).trim());
+        if(record == null) throw new IllegalArgumentException("Unknown chromosome in \""+region+"\".");
+        final int hyphen = region.indexOf('-',colon+1);
+        final int plus = region.indexOf('+',colon+1);
+        //CHROM:POS
+        if(hyphen == -1 && plus == -1) {
+            final int pos = parseCoordinate(region.substring(colon+1));
+            if( pos < 1 || pos > record.getSequenceLength()) throw new IllegalArgumentException("In "+region +", position="+pos+" out of bounds 1<=pos<="+record.getSequenceLength());
+            return new Interval( record.getSequenceName(),pos,pos);
+        }
+        //CHROM:POS+NUM
+        if(hyphen == -1 && plus > colon) {
+            final int pos = parseCoordinate(region.substring(colon+1,plus));
+            final int extend = parseCoordinate(region.substring(plus+1));
+            if( extend < 0)  throw new IllegalArgumentException("Extend cannot be negative in "+region);
+            if( pos - extend  > record.getSequenceLength() || pos+extend  <1 ) throw new IllegalArgumentException("In "+region +", position="+pos+"+/-"+extend+" out of bounds 1<=pos<="+record.getSequenceLength());
+            return new Interval( record.getSequenceName(),
+                    Math.max(1,pos-extend),
+                    Math.min(pos+extend,record.getSequenceLength())
+                    );
+        }
+      //CHROM:START-END or CHROM:START-
+       final int chromStart = parseCoordinate(region.substring(colon+1,hyphen));
+       if(chromStart > record.getSequenceLength()) throw new IllegalArgumentException("In "+region +", position="+chromStart+" out of bounds 1<=pos<="+record.getSequenceLength());
+       //CHROM:START-
+       if( region.substring(hyphen+1).trim().isEmpty()) {
+           return new Interval( record.getSequenceName(),
+                   Math.max(1,chromStart),
+                   record.getSequenceLength()
+                   );
+           }
+       //CHROM:START-END
+       final int chromEnd = parseCoordinate(region.substring(hyphen+1));
+       if(chromEnd < chromStart)  throw new IllegalArgumentException("In "+region +", start>end");
```

Comment:
Interval actually supports the end being 1 less than the start to indicate an empty interval between the two bases:

1:10-9 can point to the place of an insertion between the 9th and 10th bases. Would you like to allow for that in the parsing?